js/product/productController.js

```diff
@@ -0,0 +1,100 @@
+firmApp.controller('productCtrl', function ($scope, $http, $uibModal, $filter, localModelProduct, localRepositoryProduct) {
+
+    $scope.productList = [];
+
+    localRepositoryProduct.getDataProduct (function(data){
+        $scope.productList = data;
+    });
+
+    $scope.propertyName = 'nazwa';
+    $scope.reverse = true;
+
+    $scope.sortBy = function(propertyName) {
+        $scope.reverse = ($scope.propertyName === propertyName) ? !$scope.reverse : false;
+        $scope.propertyName = propertyName;
+    };
+
+
+    $scope.addDataProduct = function(){
+        openDialogAddProduct();
+    };
+
+    function openDialogAddProduct(){
+        var modalInstance = $uibModal.open({
+            animation: 'true',
+            templateUrl: 'addProductModal.html',
+            controller: 'addProductCtrl',
+            size: 'lg'
+        });
+
+        modalInstance.result.then(function (newProduct){
+            $scope.productList.push(newProduct);
+        });
+    };
+
+
+    $scope.editDataProduct = function(product){
+        openDialogDescriptionProduct(product);
+    };
+
+    function openDialogDescriptionProduct(product){
+        var modalInstance = $uibModal.open({
+            animation: 'true',
+            templateUrl: 'modalProductEdit.html',
+            controller: 'editProductCtrl',
+            size: 'lg',
+            resolve: {
+                product: function(){
+                    return product;
+                }
+            }
+        });
+    };
+
+    $scope.deleteDataProduct = function(idData, product){
+        localRepositoryProduct.deleteDataProduct(idData);
+
+        $scope.productList.splice($scope.productList.indexOf(product), 1);
+    }
+
+});
+
+
+firmApp.controller('editProductCtrl', function ($scope, $http, $uibModalInstance, product, localModelProduct, localRepositoryProduct) {
+    $scope.product = product;
+    $scope.newName = $scope.product.nazwa;
```

Comment:
to samo co w firmach - nie rob kopii wartosci pol do scope, tylko wyswietlaj bezposrednio ze scope'a w html.